pkg/deployer/changedetector/x509.go

```diff
@@ -0,0 +1,44 @@
+package changedetector
+
+import (
+	spinnakerv1alpha1 "github.com/armory-io/spinnaker-operator/pkg/apis/spinnaker/v1alpha1"
+	"github.com/armory-io/spinnaker-operator/pkg/halconfig"
+	"github.com/armory-io/spinnaker-operator/pkg/util"
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"strconv"
+)
+
+type x509ChangeDetector struct {
+	client client.Client
+	log    logr.Logger
+}
+
+type x509ChangeDetectorGenerator struct {
+}
+
+func (g *x509ChangeDetectorGenerator) NewChangeDetector(client client.Client, log logr.Logger) (ChangeDetector, error) {
+	return &x509ChangeDetector{client: client, log: log}, nil
+}
+
+// IsSpinnakerUpToDate returns true if there is a x509 configuration with a matching service
+func (ch *x509ChangeDetector) IsSpinnakerUpToDate(spinSvc *spinnakerv1alpha1.SpinnakerService, config runtime.Object, hc *halconfig.SpinnakerConfig) (bool, error) {
+	// ignore error as default.apiPort may not exist
+	apiPort, _ := hc.GetServiceConfigPropString("gate", "default.apiPort")
+	if apiPort == "" {
```

Comment:
If there's no apiPort (but still with `expose` option being set), we may want to remove a `Service` previously set for the x509.

Reply:
Maybe better to check if no apiPort, and a service named `spin-gate-x509` exists

Reply:
Sounds good to me. Also, the whole check should only happen if `expose.type == LoadBalancer`.